### PR TITLE
Add sorting by length for attributes

### PR DESCRIPTION
**Pull request by nikbondarenko**
Hi!
I am working on the UWP application (began from Windows 8.0 and Windows Phone 7.5). So I am spent much times for work with XAML. A few month ago I found your extension and I really like it:) Thanks!

But I had one small problem (I am a perfectionist) - not all properties sorted by length, because "Brush" property can be "Red" or "{StaticResource RedBrush}". So.. I hope you understand me :)

I add "Sort by length" property ('false' by default) and make small refactor. 
Thanks!

### CODE REVIEW

File: XamlStyler.Core/DocumentProcessors/CommentDocumentProcessor.cs [hunk 1]
```diff
@@ -73,13 +71,12 @@ public void Process(XmlReader xmlReader, StringBuilder output, ElementProcessCon
...
-                output
-                    .Append(currentIndentString)
-                    .Append("<!--")
-                    .Append(' ', this.options.CommentSpaces)
-                    .Append(content.Trim())
-                    .Append(' ', this.options.CommentSpaces)
-                    .Append("-->");
+                output.Append(currentIndentString)
+                      .Append("<!--")
+                      .Append(' ', this.options.CommentSpaces)
```
Comment (grochocki) [inline XamlStyler.Core/DocumentProcessors/CommentDocumentProcessor.cs]: A bunch of these changes look like they are not related to the new option. Can you revert these non-essential changes?

**Comment by grochocki** [comment]
Thanks for the contribution! Is sorting by length something that is done often? I imagine it is more aesthetically pleasing to some, but perhaps sacrifices consistent ordering (i.e., sometimes Foreground will be first, and sometimes it will be last, depending on the value).

**Comment by nikbondarenko** [comment]
@grochocki, for me and my team - yes, we do it often, because for me this XAML looks more clearly and more readable. Of course for many devs it can be bad, because they want see a "Foreground" property at a first line, for this case it is "False" by default :)

File: XamlStyler.Core/DocumentProcessors/CommentDocumentProcessor.cs (hunk 1 above)
Comment (nikbondarenko) [inline XamlStyler.Core/DocumentProcessors/CommentDocumentProcessor.cs]: @grochocki, ok, I will revert these changes

**Comment by grochocki** [comment]
@nikbondarenko, thanks, with false as the default, this seems like a reasonable feature to add. How do you handle attributes where the ordering matters (e.g., [Command/CommandParameter](http://www.nickdarnell.com/xaml-attribute-order-matters/), [SelectedItem/ItemSource](https://weblogs.asp.net/dixin/the-order-issue-of-xaml-attributes), etc)? My concern is that if we strictly sort by length, we may inadvertently break folks XAML.

**Comment by nikbondarenko** [comment]
@grochocki, I develop a UWP applications and I never seen troubles due to sorting (only in custom extensions). But I can err, it's only my experience. 
But as for me - if you don't want sorting by length - you disable this property :) 
When I am working with XAML - I sort it only in new files or when I open old project and start fix bugs/do refactoring.

**Comment by tziemek** [comment]
@nikbondarenko How is your experience with source control (diff, merge) with your change? I personally find it very annoying if changes of a value also results in reordering, especially if I have changed multiple values.

**Comment by nikbondarenko** [comment]
@tziemek, I have experience with `tfs` more than 4 years and I of course know problems with merge etc. But if you make your `xaml` a clear and ordered by length from beginning -> it will a usable. But if you will annoyed - you can disable this feature (it is disabled by default) . 
For me - it is like main rule for coding (`xaml`). Because for me (and my team) ordered properties more an easy to read. You start read attributes from top (less) to bottom (larger). 
And now (when I haven't this feature in extension yet) I make this manually from beginning and it's never let me annoyed.

**Comment by grochocki** [comment]
> Because for me (and my team) ordered properties more an easy to read. You start read attributes from top (less) to bottom (larger).

With any decent number of attributes, though, you lose predictability of where to look, which to me, sounds very inefficient. For example, something that is a pretty standard first attribute is something like "x:Name" or "Name", but if ordered by length, it could be anywhere in the ordering, making it more difficult to locate. If it works for your team, I can respect that, but I suppose my concern is that if it is not a standard/popular styling technique, I am not sure I am a fan of "promoting" it by making it an option.

> I develop a UWP applications and I never seen troubles due to sorting (only in custom extensions). But I can err, it's only my experience.

Attributes where the ordering matters is still my biggest concern, since the effects of this are real. Bugs as a result of specific ordering requirements are not always obvious, and introducing an option that does not address this may introduce bugs unknowingly into code.

What if, instead of ordering all attributes by length, the "*:*" and "*" wildcards are sorted by length? Right now, they are simply ordered alphabetically, so the setting becomes a "Wildcard Ordering" option where the options are "Alphabetical" (default) and "Length". That combine with this attribute ordering rule group:

```json
            "x:Class, xmlns, xmlns:x, xmlns:*",
            "x:Key, Key, x:Name, Name, x:Uid, Uid, Title",
            "*:*, *",
            "CommandParameter, Command, ItemsSource, SelectedItem"
```

would provide the ability for some amount of standard ordering and protect attributes where ordering matters, while still respecting the aesthetics of ordering by length. This rule should not apply to wildcards of other types (e.g., "AutomationProperties.*"). If you still truly want all attributes to be ordered by length, simply specify:

```json
            "*:*, *",
```

**Comment by grochocki** [comment]
After a bit more investigation and discussion I've had regarding the idea of sorting by length with various folks, we decided not to take this due to many of the reasons already listed.

**Pull request closed without merging**